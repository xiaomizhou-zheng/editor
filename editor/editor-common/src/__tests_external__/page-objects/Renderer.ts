import { InProductTestPageObject } from '@atlaskit/in-product-testing';

export class RendererPageObject extends InProductTestPageObject {
  public getContent() {
    return this.cy.get('.ak-renderer-document');
  }

  public expectContentReady() {
    return this.cy.get('.ak-renderer-document').should('be.visible');
  }
}
