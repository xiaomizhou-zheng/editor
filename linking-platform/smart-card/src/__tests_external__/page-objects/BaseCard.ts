import {
  InProductTestPageObject,
  CypressType,
} from '@atlaskit/in-product-testing';

import { CardType, CardAppearance } from '@atlaskit/linking-common';
export class BaseCardPageObject extends InProductTestPageObject {
  protected constructor(
    protected cy: CypressType,
    protected appearance: CardAppearance,
  ) {
    super(cy);
  }
  private static CARD_TIMEOUT = 15000;

  private _toSelector(appearance: CardAppearance, status: CardType) {
    const viewSelector = status.replace(/_/g, '-');
    return this.toTestId(`${appearance}-card-${viewSelector}-view`);
  }

  public expectCardReady(
    numOfCards: number = 1,
    status: CardType = 'resolved',
  ) {
    const selector = this._toSelector(this.appearance, status);

    return (
      this.cy
        // NOTE: `timeout` used since Smart Links rely on a network response.
        .get(selector, { timeout: BaseCardPageObject.CARD_TIMEOUT })
        .should('have.length', numOfCards)
        .each((element: any) => {
          this.cy
            .wrap(element)
            // NOTE: `be.visible` is used to ensure the file is still visible
            // whilst it is being uploaded.
            .should('be.visible');
        })
    );
  }

  public expectCardNotExists(status: CardType = 'resolved') {
    const selector = this._toSelector(this.appearance, status);
    return this.cy.get(selector).should('not.exist');
  }

  public findCardNth(nth: number = 1, status: CardType = 'resolved') {
    const selector = this._toSelector(this.appearance, status);
    return (
      this.cy
        // NOTE: `timeout` used since Smart Links rely on a network response.
        .get(selector, { timeout: BaseCardPageObject.CARD_TIMEOUT })
        .should('have.length.at.least', nth)
        .should('be.visible')
        .eq(nth - 1)
    );
  }

  public assertHrefRendered(url: string) {
    return this.cy
      .get(`a[href="${url}"]`, { timeout: BaseCardPageObject.CARD_TIMEOUT })
      .should('be.visible');
  }
  public assertHrefNotRendered(url: string) {
    return this.cy
      .get(`a[href="${url}"]`, { timeout: BaseCardPageObject.CARD_TIMEOUT })
      .should('not.exist');
  }

  public click(status: CardType = 'resolved') {
    return this.findCardNth(1, status).click();
  }
}
