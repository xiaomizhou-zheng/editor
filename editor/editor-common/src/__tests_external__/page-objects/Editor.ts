import { InProductTestPageObject } from '@atlaskit/in-product-testing';

import { EditorTestCaseOpts } from '../cases/types';

import { RendererPageObject } from './Renderer';

export class EditorPageObject extends InProductTestPageObject {
  public getEditorArea() {
    return this.cy.get('div.ProseMirror', { timeout: 30000 });
  }

  public getTitleArea() {
    return this.cy.get('[data-test-id="editor-title"]');
  }

  public publish(opts: EditorTestCaseOpts['ui']) {
    if (opts?.publishButton) {
      this.cy.get(opts.publishButton).click();
      return new RendererPageObject(this.cy);
    }
    throw Error('No publish button selector supplied!');
  }
}
