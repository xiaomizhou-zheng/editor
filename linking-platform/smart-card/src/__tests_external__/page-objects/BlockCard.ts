import { CypressType } from '@atlaskit/in-product-testing';

import { BaseCardPageObject } from './BaseCard';

export class BlockCardPageObject extends BaseCardPageObject {
  public constructor(protected cy: CypressType) {
    super(cy, 'block');
  }
}
