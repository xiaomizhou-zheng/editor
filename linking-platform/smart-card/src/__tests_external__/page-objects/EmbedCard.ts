import { CypressType } from '@atlaskit/in-product-testing';

import { BaseCardPageObject } from './BaseCard';

export class EmbedCardPageObject extends BaseCardPageObject {
  public constructor(protected cy: CypressType) {
    super(cy, 'embed');
  }
}
