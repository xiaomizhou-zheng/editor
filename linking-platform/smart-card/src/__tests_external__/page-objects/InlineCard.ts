import { CypressType } from '@atlaskit/in-product-testing';

import { BaseCardPageObject } from './BaseCard';

export class InlineCardPageObject extends BaseCardPageObject {
  public constructor(protected cy: CypressType) {
    super(cy, 'inline');
  }
}
