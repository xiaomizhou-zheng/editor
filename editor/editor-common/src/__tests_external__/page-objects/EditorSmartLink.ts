import {
  CypressType,
  InProductTestPageObject,
} from '@atlaskit/in-product-testing';
import { CardAppearance } from '@atlaskit/smart-card';
import {
  BlockCardPageObject,
  EmbedCardPageObject,
  InlineCardPageObject,
} from '@atlaskit/smart-card/in-product';

import { EditorPageObject } from './Editor';

export class EditorSmartLinkPageObject extends InProductTestPageObject {
  constructor(protected cy: CypressType, private editor: EditorPageObject) {
    super(cy);
  }

  testIds = {
    viewSwitcher: 'link-toolbar-appearance-button',
    displayUrlOption: 'url-appearance',
    displayInlineOption: 'inline-appearance',
    displayCardOption: 'block-appearance',
    displayEmbedOption: 'embed-appearance',
    linkUrl: 'link-url',
    linkLabel: 'link-label',
  };
  ariaLabels = {
    editLink: 'Edit link',
    deleteLink: 'Remove',
    unlinkLink: 'Unlink',
  };

  private getViewSwitcher() {
    return this.cy.get(this.toTestId(this.testIds.viewSwitcher));
  }

  public openViewSwitcher() {
    return this.getViewSwitcher().click();
  }

  private getViewSwitcherOption(type: CardAppearance | 'url' = 'inline') {
    switch (type) {
      case 'url': {
        return this.cy.get(this.toTestId(this.testIds.displayUrlOption));
      }
      case 'inline': {
        return this.cy.get(this.toTestId(this.testIds.displayInlineOption));
      }
      case 'block': {
        return this.cy.get(this.toTestId(this.testIds.displayCardOption));
      }
      case 'embed': {
        return this.cy.get(this.toTestId(this.testIds.displayEmbedOption));
      }
      default: {
        throw Error(
          `Attempted to switch to \`${type}\`: unknown Smart Link view switcher option!`,
        );
      }
    }
  }

  public selectViewSwitcherOption(type: CardAppearance | 'url' = 'inline') {
    const viewSwitcherOption = this.getViewSwitcherOption(type);
    return viewSwitcherOption.click();
  }

  public insertSmartLinkByTyping(url: string) {
    return this.editor
      .getEditorArea()
      .type(`${url} `, { delay: 0 })
      .get('p>a')
      .type('{leftArrow}{leftArrow}')
      .get('[aria-label="Floating Toolbar"]')
      .then(() => {
        this.openViewSwitcher();
        this.selectViewSwitcherOption('inline');
      });
  }

  public switchAfterInsert(type: CardAppearance | 'url' = 'inline') {
    // NOTE: inline inserted by default for all Smart Links.
    const inlineCard = new InlineCardPageObject(this.cy);
    inlineCard.click();
    this.openViewSwitcher();
    this.selectViewSwitcherOption(type);
  }

  private getSmartLink(type: CardAppearance | 'url' = 'inline') {
    switch (type) {
      case 'inline': {
        return new InlineCardPageObject(this.cy);
      }
      case 'block': {
        return new BlockCardPageObject(this.cy);
      }
      case 'embed': {
        return new EmbedCardPageObject(this.cy);
      }
      case 'url': {
        return new InlineCardPageObject(this.cy);
      }
    }
  }

  public getEditLinkButton() {
    return this.cy.get(this.toAriaLabel(this.ariaLabels.editLink));
  }

  public openEditLinkMenu() {
    const button = this.getEditLinkButton();
    return button.click();
  }

  public changeLinkLabel(
    title: string,
    type: CardAppearance | 'url' = 'inline',
  ) {
    const smartLink = this.getSmartLink(type);
    smartLink.click();
    this.openEditLinkMenu();
    this.typeIntoLabelField(title);
  }

  public changeLinkUrl(url: string, type: CardAppearance | 'url' = 'inline') {
    this.getSmartLink(type).click();
    this.openEditLinkMenu();
    this.typeIntoUrlField(url);
  }

  private typeIntoLabelField(title: string) {
    const titleField = this.cy
      .get(this.toTestId(this.testIds.linkLabel))
      .focus();
    titleField.clear();
    titleField.type(`${title}{enter}`, { delay: 0 });
  }
  private typeIntoUrlField(url: string) {
    const urlField = this.cy.get(this.toTestId(this.testIds.linkUrl)).focus();
    urlField.clear();
    urlField.type(`${url}{enter}`, { delay: 0 });
  }

  public deleteSmartLink(type: CardAppearance | 'url' = 'inline') {
    this.getSmartLink(type).click();
    this.clickDeleteIcon();
  }
  private clickDeleteIcon() {
    const ariaLabelSelector = this.toAriaLabel(this.ariaLabels.deleteLink);
    const deleteButtonSelector = `button${ariaLabelSelector}`;
    return this.cy.get(deleteButtonSelector).click();
  }

  public unlink(type: CardAppearance | 'url' = 'inline') {
    this.getSmartLink(type).click();
    this.clickUnlinkIcon();
  }
  private clickUnlinkIcon() {
    const ariaLabelSelector = this.toAriaLabel(this.ariaLabels.unlinkLink);
    const unlinkButtonSelector = `button${ariaLabelSelector}`;
    return this.cy.get(unlinkButtonSelector).click();
  }
}
