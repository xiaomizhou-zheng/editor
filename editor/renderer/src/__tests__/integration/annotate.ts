import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  mountRenderer,
  goToRendererTestingExample,
} from '../__helpers/testing-example-helpers';
import { selectors } from '../__helpers/page-objects/_renderer';
import { annotate } from '../__helpers/page-objects/_actions';
import taskDecisionAdf from './__fixtures__/task-decision.adf.json';

const CHAR_WIDTH = 6;
// TODO: https://product-fabric.atlassian.net/browse/ED-9831
// Selection in Catalina Safari isn't working properly.
BrowserTestCase(
  `Can't create an annotation on a text selection that contains inline nodes`,
  { skip: ['firefox', 'safari'] },
  async (client: any) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(
      page,
      { withRendererActions: true },
      {
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'hello ',
              },
              {
                type: 'emoji',
                attrs: {
                  shortName: ':cactus:',
                  id: '1f335',
                  text: 'ðŸŒµ',
                },
              },
              {
                type: 'text',
                text: ' mate',
              },
            ],
          },
        ],
      },
    );

    const selector = `${selectors.document} > p`;
    await page.waitForSelector(selector);
    await page.simulateUserSelection(selector, selector, (element) =>
      element === 'start' ? CHAR_WIDTH : 0,
    );

    const result = await annotate(page, '1234');
    expect(result).toEqual(false);
  },
);

BrowserTestCase(
  `Can't create an annotation on a text selection that falls in the middle of an inline node`,
  { skip: ['firefox', 'safari'] },
  async (client: any) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(
      page,
      { withRendererActions: true },
      {
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'emoji',
                attrs: {
                  shortName: ':not-an-emoji:',
                  id: '',
                  text: '',
                },
              },
              {
                type: 'text',
                text: 'hello ',
              },
            ],
          },
        ],
      },
    );

    const selector = `${selectors.document} > p`;
    await page.waitForSelector(selector);
    await page.simulateUserSelection(selector, selector, (element) =>
      element === 'start' ? CHAR_WIDTH * 3 : 0,
    );

    const result = await annotate(page, '1234');
    expect(result).toEqual(false);
  },
);

BrowserTestCase(
  'Can create an annotation on a basic text selection',
  { skip: ['firefox', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(
      page,
      { withRendererActions: true },
      {
        type: 'doc',
        version: 1,
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'Hello, World',
              },
            ],
          },
        ],
      },
    );

    const selector = `${selectors.document} > p`;

    await page.waitForSelector(selector);
    await page.simulateUserSelection(selector, selector, (element) =>
      element === 'start' ? CHAR_WIDTH : 0,
    );

    const result = await annotate(page, '1234');
    expect(result).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'Can create an annotation on a text selection over two paragraphs',
  { skip: ['firefox', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(
      page,
      { withRendererActions: true },
      {
        type: 'doc',
        version: 1,
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text:
                  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean mi nisl, venenatis eget auctor vitae, venenatis quis lorem. Suspendisse maximus tortor vel dui tincidunt cursus. Vestibulum magna nibh, auctor non auctor id, finibus vitae orci. Nulla viverra ipsum et nunc fringilla ultricies. Pellentesque vitae felis molestie justo finibus accumsan. Suspendisse potenti. Nulla facilisi. Integer dignissim quis velit quis elementum. Sed sit amet varius ante. Duis vestibulum porta augue eu laoreet. Morbi id risus et augue sollicitudin aliquam. In et ligula dolor. Nam ac aliquet diam.',
              },
            ],
          },
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text:
                  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean mi nisl, venenatis eget auctor vitae, venenatis quis lorem. Suspendisse maximus tortor vel dui tincidunt cursus. Vestibulum magna nibh, auctor non auctor id, finibus vitae orci. Nulla viverra ipsum et nunc fringilla ultricies. Pellentesque vitae felis molestie justo finibus accumsan. Suspendisse potenti. Nulla facilisi. Integer dignissim quis velit quis elementum. Sed sit amet varius ante. Duis vestibulum porta augue eu laoreet. Morbi id risus et augue sollicitudin aliquam. In et ligula dolor. Nam ac aliquet diam.',
              },
            ],
          },
        ],
      },
    );

    const selector = `${selectors.document} > p`;

    await page.waitForSelector(selector);
    await page.simulateUserSelection(
      selector,
      `${selector}:nth-child(2)`,
      (element) => (element === 'start' ? CHAR_WIDTH : -(CHAR_WIDTH * 5)),
    );

    const result = await annotate(page, '1234');
    expect(result).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'Can create an annotation on a text selection over a decision item',
  { skip: ['firefox', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(page, { withRendererActions: true }, taskDecisionAdf);

    const selector = `${selectors.document} [data-decision-local-id] [data-renderer-start-pos]`;

    await page.waitForSelector(selector);
    await page.simulateUserSelection(selector, selector, (element) =>
      element === 'start' ? CHAR_WIDTH : 0,
    );

    const result = await annotate(page, '1234');
    expect(result).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'Can create an annotation on a text selection over a task item',
  { skip: ['firefox', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(page, { withRendererActions: true }, taskDecisionAdf);

    const selector = `${selectors.document} [data-task-local-id] [data-renderer-start-pos]`;

    await page.waitForSelector(selector);
    await page.simulateUserSelection(selector, selector, (element) =>
      element === 'start' ? CHAR_WIDTH : 0,
    );

    const result = await annotate(page, '1234');
    expect(result).toMatchCustomDocSnapshot(testName);
  },
);
