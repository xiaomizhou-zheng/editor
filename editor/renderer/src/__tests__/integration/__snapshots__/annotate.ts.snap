// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can create an annotation on a basic text selection 1`] = `
Object {
  "doc": Object {
    "content": Array [
      Object {
        "content": Array [
          Object {
            "text": "H",
            "type": "text",
          },
          Object {
            "marks": Array [
              Object {
                "attrs": Object {
                  "annotationType": "inlineComment",
                  "id": "1234",
                },
                "type": "annotation",
              },
            ],
            "text": "ello, World",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "doc",
    "version": 1,
  },
  "step": Object {
    "from": 2,
    "mark": Object {
      "attrs": Object {
        "annotationType": "inlineComment",
        "id": "1234",
      },
      "type": "annotation",
    },
    "stepType": "addMark",
    "to": 13,
  },
}
`;

exports[`Can create an annotation on a text selection over a decision item 1`] = `
Object {
  "doc": Object {
    "content": Array [
      Object {
        "content": Array [
          Object {
            "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean mi nisl, venenatis eget auctor vitae, venenatis quis lorem. Suspendisse maximus tortor vel dui tincidunt cursus. Vestibulum magna nibh, auctor non auctor id, finibus vitae orci. ",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
      Object {
        "attrs": Object {
          "localId": "4b895470-4b94-4c9c-84ac-dbebfec31b2a",
        },
        "content": Array [
          Object {
            "attrs": Object {
              "localId": "39da70eb-1e13-4d75-857f-7112943f548d",
              "state": "TODO",
            },
            "content": Array [
              Object {
                "text": "hello, mate",
                "type": "text",
              },
            ],
            "type": "taskItem",
          },
        ],
        "type": "taskList",
      },
      Object {
        "attrs": Object {
          "localId": "b8510754-475d-4ee4-9887-d090024a21be",
        },
        "content": Array [
          Object {
            "attrs": Object {
              "localId": "180d58fe-a7bf-48fb-8d34-3bb743a07a08",
              "state": "DECIDED",
            },
            "content": Array [
              Object {
                "text": "h",
                "type": "text",
              },
              Object {
                "marks": Array [
                  Object {
                    "attrs": Object {
                      "annotationType": "inlineComment",
                      "id": "1234",
                    },
                    "type": "annotation",
                  },
                ],
                "text": "ello, mate",
                "type": "text",
              },
            ],
            "type": "decisionItem",
          },
        ],
        "type": "decisionList",
      },
      Object {
        "attrs": Object {
          "isNumberColumnEnabled": false,
          "layout": "default",
        },
        "content": Array [
          Object {
            "content": Array [
              Object {
                "attrs": Object {},
                "content": Array [
                  Object {
                    "attrs": Object {
                      "localId": "1d944f46-ca2c-4d79-b0f9-0c50945c1d95",
                    },
                    "content": Array [
                      Object {
                        "attrs": Object {
                          "localId": "ac50aece-412f-4214-b39d-c1f2945ab12a",
                          "state": "TODO",
                        },
                        "content": Array [
                          Object {
                            "marks": Array [
                              Object {
                                "type": "em",
                              },
                              Object {
                                "type": "strong",
                              },
                              Object {
                                "type": "underline",
                              },
                            ],
                            "text": "bold ",
                            "type": "text",
                          },
                          Object {
                            "marks": Array [
                              Object {
                                "attrs": Object {
                                  "href": "https://product-fabric.atlassian.net/browse/FAB-1520",
                                },
                                "type": "link",
                              },
                              Object {
                                "type": "em",
                              },
                              Object {
                                "type": "strong",
                              },
                              Object {
                                "type": "underline",
                              },
                            ],
                            "text": "FAB-1520 UI: Poor man's search",
                            "type": "text",
                          },
                        ],
                        "type": "taskItem",
                      },
                    ],
                    "type": "taskList",
                  },
                ],
                "type": "tableCell",
              },
            ],
            "type": "tableRow",
          },
        ],
        "type": "table",
      },
      Object {
        "content": Array [],
        "type": "paragraph",
      },
    ],
    "type": "doc",
    "version": 1,
  },
  "step": Object {
    "from": 262,
    "mark": Object {
      "attrs": Object {
        "annotationType": "inlineComment",
        "id": "1234",
      },
      "type": "annotation",
    },
    "stepType": "addMark",
    "to": 272,
  },
}
`;

exports[`Can create an annotation on a text selection over a task item 1`] = `
Object {
  "doc": Object {
    "content": Array [
      Object {
        "content": Array [
          Object {
            "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean mi nisl, venenatis eget auctor vitae, venenatis quis lorem. Suspendisse maximus tortor vel dui tincidunt cursus. Vestibulum magna nibh, auctor non auctor id, finibus vitae orci. ",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
      Object {
        "attrs": Object {
          "localId": "4b895470-4b94-4c9c-84ac-dbebfec31b2a",
        },
        "content": Array [
          Object {
            "attrs": Object {
              "localId": "39da70eb-1e13-4d75-857f-7112943f548d",
              "state": "TODO",
            },
            "content": Array [
              Object {
                "text": "h",
                "type": "text",
              },
              Object {
                "marks": Array [
                  Object {
                    "attrs": Object {
                      "annotationType": "inlineComment",
                      "id": "1234",
                    },
                    "type": "annotation",
                  },
                ],
                "text": "ello, mate",
                "type": "text",
              },
            ],
            "type": "taskItem",
          },
        ],
        "type": "taskList",
      },
      Object {
        "attrs": Object {
          "localId": "b8510754-475d-4ee4-9887-d090024a21be",
        },
        "content": Array [
          Object {
            "attrs": Object {
              "localId": "180d58fe-a7bf-48fb-8d34-3bb743a07a08",
              "state": "DECIDED",
            },
            "content": Array [
              Object {
                "text": "hello, mate",
                "type": "text",
              },
            ],
            "type": "decisionItem",
          },
        ],
        "type": "decisionList",
      },
      Object {
        "attrs": Object {
          "isNumberColumnEnabled": false,
          "layout": "default",
        },
        "content": Array [
          Object {
            "content": Array [
              Object {
                "attrs": Object {},
                "content": Array [
                  Object {
                    "attrs": Object {
                      "localId": "1d944f46-ca2c-4d79-b0f9-0c50945c1d95",
                    },
                    "content": Array [
                      Object {
                        "attrs": Object {
                          "localId": "ac50aece-412f-4214-b39d-c1f2945ab12a",
                          "state": "TODO",
                        },
                        "content": Array [
                          Object {
                            "marks": Array [
                              Object {
                                "type": "em",
                              },
                              Object {
                                "type": "strong",
                              },
                              Object {
                                "type": "underline",
                              },
                            ],
                            "text": "bold ",
                            "type": "text",
                          },
                          Object {
                            "marks": Array [
                              Object {
                                "attrs": Object {
                                  "href": "https://product-fabric.atlassian.net/browse/FAB-1520",
                                },
                                "type": "link",
                              },
                              Object {
                                "type": "em",
                              },
                              Object {
                                "type": "strong",
                              },
                              Object {
                                "type": "underline",
                              },
                            ],
                            "text": "FAB-1520 UI: Poor man's search",
                            "type": "text",
                          },
                        ],
                        "type": "taskItem",
                      },
                    ],
                    "type": "taskList",
                  },
                ],
                "type": "tableCell",
              },
            ],
            "type": "tableRow",
          },
        ],
        "type": "table",
      },
      Object {
        "content": Array [],
        "type": "paragraph",
      },
    ],
    "type": "doc",
    "version": 1,
  },
  "step": Object {
    "from": 247,
    "mark": Object {
      "attrs": Object {
        "annotationType": "inlineComment",
        "id": "1234",
      },
      "type": "annotation",
    },
    "stepType": "addMark",
    "to": 257,
  },
}
`;

exports[`Can create an annotation on a text selection over two paragraphs 1`] = `
Object {
  "doc": Object {
    "content": Array [
      Object {
        "content": Array [
          Object {
            "text": "L",
            "type": "text",
          },
          Object {
            "marks": Array [
              Object {
                "attrs": Object {
                  "annotationType": "inlineComment",
                  "id": "1234",
                },
                "type": "annotation",
              },
            ],
            "text": "orem ipsum dolor sit amet, consectetur adipiscing elit. Aenean mi nisl, venenatis eget auctor vitae, venenatis quis lorem. Suspendisse maximus tortor vel dui tincidunt cursus. Vestibulum magna nibh, auctor non auctor id, finibus vitae orci. Nulla viverra ipsum et nunc fringilla ultricies. Pellentesque vitae felis molestie justo finibus accumsan. Suspendisse potenti. Nulla facilisi. Integer dignissim quis velit quis elementum. Sed sit amet varius ante. Duis vestibulum porta augue eu laoreet. Morbi id risus et augue sollicitudin aliquam. In et ligula dolor. Nam ac aliquet diam.",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
      Object {
        "content": Array [
          Object {
            "marks": Array [
              Object {
                "attrs": Object {
                  "annotationType": "inlineComment",
                  "id": "1234",
                },
                "type": "annotation",
              },
            ],
            "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean mi nisl, venenatis eget auctor vitae, venenatis quis lorem. Suspendisse maximus tortor vel dui tincidunt cursus. Vestibulum magna nibh, auctor non auctor id, finibus vitae orci. Nulla viverra ipsum et nunc fringilla ultricies. Pellentesque vitae felis molestie justo finibus accumsan. Suspendisse potenti. Nulla facilisi. Integer dignissim quis velit quis elementum. Sed sit amet varius ante. Duis vestibulum porta augue eu laoreet. Morbi id risus et augue sollicitudin aliquam. In et ligula dolor. Nam ac aliquet diam.",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "doc",
    "version": 1,
  },
  "step": Object {
    "from": 2,
    "mark": Object {
      "attrs": Object {
        "annotationType": "inlineComment",
        "id": "1234",
      },
      "type": "annotation",
    },
    "stepType": "addMark",
    "to": 1169,
  },
}
`;
